documents/invoice.md

```diff
@@ -0,0 +1,529 @@
+## Invoices
+
+### Create Invoice
+
+Request #1
+In this example, an invoice is created using the customer and item details. Here, the customer and item are created while creating the invoice.
+```go
+
+line_items := make(map[string]interface{})
+line_items["0"] = map[string]interface{}{
+      "name": "Master Cloud Computing in 30 Days",
+      "description": "Book by Ravena Ravenclaw",
+      "amount": 399,
+      "currency": "USD",
+      "quantity": 1,
+    }
+
+data := map[string]interface{}{
+  "type": "invoice",
+  "description": "Invoice for the month of January 2020",
+  "partial_payment": true,
+  "customer": map[string]interface{}{
+    "name": "Gaurav Kumar",
+    "contact": 9999999999,
+    "email": "gaurav.kumar@example.com",
+    "billing_address": {
+      "line1": "Ground & 1st Floor, SJR Cyber Laskar",
+      "line2": "Hosur Road",
+      "zipcode": 560068,
+      "city": "Bengaluru",
+      "state": "Karnataka",
+      "country": "in",
+    },
+    "shipping_address": map[string]interface{}{
+      "line1": "Ground & 1st Floor, SJR Cyber Laskar",
+      "line2": "Hosur Road",
+      "zipcode": 560068,
+      "city": "Bengaluru",
+      "state": "Karnataka",
+      "country": "in",
+    },
+  },
+  "line_items": line_items,
+  "sms_notify": 1,
+  "email_notify": 1,
+  "currency": "USD",
+  "expire_by": 1589765167
+}
+body, err := client.Invoice.Create(data, nil)
+```
+
+Request #2
+In this example, an invoice is created using existing `customer_id` and `item_id`
+```go
+line_items := make(map[string]interface{})
+line_items["0"] = map[string]interface{}{
+      "item_id": "item_DRt61i2NnL8oy6",
+}
+
+data := map[string]interface{}{
+  "type": "invoice",
+  "date": 1589994898,
+  "customer_id": "cust_E7q0trFqXgExmT",
+  "line_items": line_items,
+}
+
+body, err := client.Invoice.Create(data, nil)
+```
+
+**Parameters:**
+
+| Name            | Type    | Description                                                                  |
+|-----------------|---------|------------------------------------------------------------------------------|
+|type*          | string | entity type (here its invoice)                                               |
+|description        | string  | A brief description of the invoice.                      |
+|customer_id           | string  | customer id for which invoice need be raised                     |
+|customer           | object  | customer details in a object format                     |
+
```

Comment:
Shouldn’t we add all the request params mentioned [here](https://razorpay.com/docs/api/payments/invoices/#request-parameters)